control-plane/agents/src/bin/core/node/wrapper.rs

```diff
@@ -1283,6 +1283,14 @@ impl ReplicaApi for Arc<tokio::sync::RwLock<NodeWrapper>> {
 
                 Ok(replica)
             }
+            Err(SvcError::GrpcRequestError { source, .. })
+                if source.code() == tonic::Code::DataLoss =>
+            {
+                Err(SvcError::ReplicaSetPropertyFailed {
```

Comment:
Why do we need this additional mapping? Can't we just keep the grpc error as is?

Reply:
It gives more context on why grpc failed with `Dataloss` status code. If it fails in `create_replica` we just mark replica for deletion.